build.gradle

```diff
@@ -1,50 +1,79 @@
-    testCompile group: 'org.slf4j', name:'slf4j-api', version:'1.7.25'
-    testCompile group: 'org.slf4j', name:'slf4j-log4j12', version:'1.7.25'
-    testCompile group: 'junit', name: 'junit', version: '4.12'
-    compile group: 'org.springframework', name: 'spring-core', version: $springVersion
-    compile group: 'org.springframework', name: 'spring-beans', version: $springVersion
-    compile group: 'org.springframework', name: 'spring-context', version: $springVersion
-    compile group: 'org.springframework.security', name: 'spring-security-bom', version: $springSecurityVersion
-    compile group: 'org.springframework.security', name: 'spring-security-web', version: $springSecurityVersion
-    compile group: 'org.springframework.security', name: 'spring-security-config', version: $springSecurityVersion
-    compile group: 'org.springframework.security', name: 'spring-security-core', version: $springSecurityVersion
+	// Each compileOnly dependency is already part of the geoserver webapp
+	compileOnly "org.geotools:gt-cql:${geotoolsVersion}"
+	compileOnly "org.geotools:gt-epsg-hsql:${geotoolsVersion}"
+	compileOnly "org.geotools:gt-jdbc:${geotoolsVersion}"
+	compileOnly "org.geotools:gt-main:${geotoolsVersion}"
+	compileOnly "org.springframework:spring-context:${springVersion}"
+	compileOnly "org.springframework.security:spring-security-bom:${springSecurityVersion}"
+	compileOnly "org.springframework.security:spring-security-core:${springSecurityVersion}"
+	compileOnly "org.springframework.security:spring-security-config:${springSecurityVersion}"
+	compileOnly "org.springframework.security:spring-security-web:${springSecurityVersion}"
+	compileOnly "org.locationtech.jts:jts-core:${jtsVersion}"
+	compileOnly "org.apache.commons:commons-collections4:4.2"
+	compileOnly "com.fasterxml.jackson.core:jackson-databind:2.13.4.2"
+
+	// These are the only dependencies needed at runtime, in addition to the compileOnly ones above that are already
+	// present in the geoserver webapp.
+	implementation "com.marklogic:marklogic-client-api:6.1.0"
+	implementation "org.geotools:gt-geojson:${geotoolsVersion}"
+
+	// Have to repeat some compileOnly dependencies so the tests can compile
+	testImplementation "org.geotools:gt-cql:${geotoolsVersion}"
+	testImplementation "org.geotools:gt-epsg-hsql:${geotoolsVersion}"
+	testImplementation "org.geotools:gt-jdbc:${geotoolsVersion}"
+	testImplementation "org.geotools:gt-main:${geotoolsVersion}"
+	testImplementation "org.springframework.security:spring-security-core:${springSecurityVersion}"
+	testImplementation "com.fasterxml.jackson.core:jackson-databind:2.13.4.2"
+	testImplementation "org.glassfish.jersey.core:jersey-client:2.25.1"
+
+	testImplementation "junit:junit:4.13.2"
+	testImplementation "ch.qos.logback:logback-classic:1.3.5"
+	testImplementation "org.slf4j:jcl-over-slf4j:1.7.36"
+	testImplementation "org.slf4j:slf4j-api:1.7.36"
 }
 
+shadowJar {
```

Comment:
Now using a shadowjar to deploy the plugin with its dependencies. And using `compileOnly` to avoid bringing in a bunch of jars that already exist in the geoserver webapp.